### PR TITLE
Restrict `Level`s to real quantities

### LINKED ISSUE #398: Complex `Level`
It is possible to create a complex `Level`:
```julia
julia> @dB (3+2im)V/1V
11.139433523068368 + 5.1073257213090795im dBV

julia> @dB (3+2im)V/(1+1im)V
8.129133566428555 - 1.714556047900126im dB ((1 + 1im) V)
```
Should this be restricted to real quantities? It is not possible to create a complex `Gain`, so this seems inconsistent (and not very useful, at least to me).

And if we want to keep complex `Level`s, the printing needs to be fixed (there need to be parentheses around the complex number).

### PR DESCRIPTION
**Pull request by sostock**
Fixes #398. This allows only real quantities in `Level`s (for the value as well as the reference quantity).

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -225,24 +225,25 @@ used in promotion to put levels and gains onto a common log scale.
 """
 abstract type LogScaled{L<:LogInfo} <: Number end
 
+const RealQuantity = Union{Real, AbstractQuantity{<:Real}}
+
 """
     struct Level{L, S, T<:Number} <: LogScaled{L}
```
Comment (giordano) [inline src/types.jl:231]: Update also the docstring?

File: src/types.jl [hunk 2]
```diff
@@ -279,6 +280,8 @@ const RootPowerRatio{T} = IsRootPowerRatio{true,T}
 @inline unwrap(x::IsRootPowerRatio) = x.val
 @inline unwrap(x) = x
 
+const ReferenceQuantity = Union{RealQuantity, IsRootPowerRatio{S,<:RealQuantity} where S}
+
```
Comment (giordano) [inline src/types.jl:284]: Maybe define before the first use?

Comment (sostock) [reply src/types.jl:284]: For that, we would need to move the definition of `IsRootPowerRatio` as well. I have no problem with that, what do you think?

File: src/types.jl (hunk 1 above)
Comment (sostock) [reply src/types.jl]: Done!

File: src/types.jl (hunk 2 above)
Comment (giordano) [reply src/types.jl:284]: Ok, yeah, it's better to not move stuff around for no good reason

**Pull request merged**